routing/pathfind.go

```diff
@@ -551,34 +551,16 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		return nil, 0, errMissingDependentFeature
 	}
 
-	// Now that we know the feature vector is well formed, we'll proceed in
-	// checking that it supports the features we need, given our
-	// restrictions on the final hop.
-
-	// If the caller needs to send custom records, check that our
-	// destination feature vector supports TLV.
-	if len(r.DestCustomRecords) > 0 &&
-		!features.HasFeature(lnwire.TLVOnionPayloadOptional) {
-
-		return nil, 0, errNoTlvPayload
-	}
-
-	// If the caller has a payment address to attach, check that our
-	// destination feature vector supports them.
+	// Now that we know the feature vector is well-formed, we'll proceed in
+	// checking that it supports the features we need. If the caller has a
+	// payment address to attach, check that our destination feature vector
+	// supports them.
 	if r.PaymentAddr != nil &&
 		!features.HasFeature(lnwire.PaymentAddrOptional) {
 
 		return nil, 0, errNoPaymentAddr
 	}
```

Comment:
think we may still fail here though?

Reply:
ok nvm - for keysend this will still be useful

Reply:
If we used a payment address (MPP/AMP) then yes. But at the moment the integration tests mostly fail during the keysend payments, which don't require a payment address.